### ISSUE TITLE
Document that Filebeat should not collect its own logs on Kubernetes

### ISSUE THREAD

**Issue by belimawr** [elastic/beats#39566]
When using Kubernetes autodiscover, the example configuration [from our docs](https://www.elastic.co/guide/en/beats/filebeat/current/running-on-kubernetes.html) will make Filebeat collect it's own logs. This is fine most of the time.

However if Filebeat is logging in debug level and the `processors` selector is enabled, this will cause Filebeat to log every event it ingested causing a loop of ever growing log entries:
1. Filebeat reads any event
2. Filebeat ingests the event and logs the event in level debug
3. Because Filebeat is also collecting its own logs, at some point it will read the log generated at 2.
4. Filebeat ingests its own log from 3, then logs it in debug level. Now the log from 2 has been "wrapped" into another log entry that is also logged
5. GOTO 2

We should at least add a warning in our documentation about it and provide an example of how to prevent Filebeat from collecting its own logs.

**Comment by cmacknz** [comment]
So the loop we have is something like:

1. Filebeat reads event A.
2. Filebeat logs that it published event A.
3. Filebeat reads the log for publishing event A.
4. Filebeat logs that it published the log for publishing event A.
5. ... 

At step 2 is there some obvious marker we can add to this log message so that in step 3 Filebeat can drop this log line, or at least not recursively log about it? Whatever marker we choose has to still work if the Filebeat pod is restarted, that is it cannot be ephemeral.

**Comment by lucabelluccini** [comment]
Isn't this true also for Elastic Agents collecting its own logs via Collect Logs & Metrics?

**Comment by belimawr** [comment]
> Isn't this true also for Elastic Agents collecting its own logs via Collect Logs & Metrics?

No, for Elastic-Agent we deploy a separate Filebeat (usually called filestream-monitoring) that collects the Elastic-Agent logs and all other components (it's a single log file), but drops it's own logs, so we don't have this problem.
